lib/http.js

```diff
@@ -61,28 +61,39 @@ function createServer(opts, requestListener) {
 
 /**
  * @typedef {object} HTTPRequestOptions
- * @property {httpAgent.Agent | boolean} [agent]
- * @property {string} [auth]
- * @property {Function} [createConnection]
- * @property {number} [defaultPort]
- * @property {number} [family]
- * @property {object} [headers]
- * @property {number} [hints]
- * @property {string} [host]
- * @property {string} [hostname]
- * @property {boolean} [insecureHTTPParser]
- * @property {string} [localAddress]
- * @property {number} [localPort]
- * @property {Function} [lookup]
- * @property {number} [maxHeaderSize]
- * @property {string} [method]
- * @property {string} [path]
- * @property {number} [port]
- * @property {string} [protocol]
- * @property {boolean} [setHost]
- * @property {string} [socketPath]
- * @property {number} [timeout]
- * @property {AbortSignal} [signal]
+ * @property {httpAgent.Agent | boolean} [agent] - Controls agent behavior.
+ * @property {string} [auth] - Basic authentication ('user:password') to compute
+ * an Authorization header.
+ * @property {Function} [createConnection] - A function that produces a
+ * socket/stream to use for the request when the `agent` option is not used.
+ * @property {number} [defaultPort] - Default port for the protocol.
+ * @property {number} [family] - IP address family to use when resolving host or
+ * hostname.
+ * @property {object} [headers] - An object containing request headers.
+ * @property {number} [hints] - Optional dns.lookup() hints.
+ * @property {string} [host] - A domain name or IP address of the server to
+ * issue the request to.
+ * @property {string} [hostname] - Alias for host.
+ * @property {boolean} [insecureHTTPParser] - Use an insecure HTTP parser that
+ * accepts invalid HTTP headers when true.
+ * @property {string} [localAddress] - Local interface to bind for network
+ * connections.
+ * @property {number} [localPort] - Local port to connect from.
+ * @property {Function} [lookup] - Custom lookup function.
+ * @property {number} [maxHeaderSize] - Optionally overrides the value of
+ * --max-http-header-size (the maximum length of response headers in bytes) for
```

Comment:
That's markdown, but this is not a markdown file. So the question is whether we want to commit to using markdown. (Will it be styled appropriately in Visual Studio Code and other IDEs?)

One upside is that it would ease a transition from these things being in markdown files to them being in JSDoc comments that then get processed and moved to the markdown files automatically.

Reply:
> (Will it be styled appropriately in Visual Studio Code and other IDEs?)

Yes. JSDoc supports markdown.

Reply:
Great, then let's go for it. I'll restore all the markdown I stripped from these comments in the first place. (Except for maybe links? Do those tend to be problematic? Guess I can test and find out.)